waspc/src/Wasp/AppSpec/ExtImport.hs

```diff
@@ -23,6 +24,24 @@ data ExtImport = ExtImport
   }
   deriving (Show, Eq, Data)
 
+instance FromJSON ExtImport where
+  parseJSON = withObject "ExtImport" $ \o -> do
+    kindStr <- o .: "kind"
+    nameStr <- o .: "name"
+    pathStr <- o .: "path"
+    extImportName <- parseExtImportName kindStr nameStr
+    extImportPath <- parseExtImportPath pathStr
+    return $ ExtImport extImportName extImportPath
+    where
+      parseExtImportName kindStr nameStr = case kindStr of
+        "default" -> pure $ ExtImportModule nameStr
+        "named" -> pure $ ExtImportField nameStr
+        _ -> fail $ "Failed to parse import kind: " <> kindStr
+
+      parseExtImportPath pathStr = case parseRelFileP pathStr of
+        Just path' -> pure path'
+        Nothing -> fail $ "Failed to parse relative posix path to file: " <> pathStr
+
 type ExtImportPath = Path Posix (Rel SourceExternalCodeDir) File'
 
 type Identifier = String
```

Comment:
Yeah I get what you mean. That is why general type classes liek Show and FromJSON are good for stuff that is inherent to the type in question, because that means their implementation does not depend on who uses it, but when implementation becomes tied to the specific usage, then it is better to have custom type classes or just functions, so it is clear.

In this case, language server has no expectations but to serialize/deserialize so that is ok.
And as for deserializing AppSpec from JSON that was created from TS SDK -> in that case also, there is no reason why TS SDK would dictate what that JSON looks like, instead TS SDK will make sure to follow the format dictated by AppSpec (and its FromJSON instance).
So that is ok -> FromJSON indeed seems to be inherent to the AppSpec here.

The only tricky thing could be if somebody happily changes how some part of AppSpec does its FromJSON, without considering that there are external producers of such JSON (like TS SDK) that would also need to be updated. But I think that is up to that person to think about.